### PR TITLE
[Chore] Setup code formatter

### PR DESCRIPTION
**Pull request by rosle**
## What happened

✅  Set up code formatter for the project

## Insight

- Editor config
- Git pre-commit hook to format files

## Proof Of Work

`N/A`

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (rafayet-monon) [inline .prettierrc:1]: Can we add the [bracketSpacing](https://prettier.io/docs/en/options.html#bracket-spacing) rule? What do you think?

**Review by rafayet-monon — COMMENTED** [review]
Have you considered using `eslint`?

File: .prettierrc (hunk 1 above)
Comment (rosle) [reply .prettierrc:1]: Added 👍

**Comment by rosle** [comment]
> Have you considered using `eslint`?

I think it has already included by default as it has some configs in package.json 🤔  I can also run the command. But I think I should add a simple github action to run the format check and eslint again 💭  Can I do it in another PR?

**Review by rafayet-monon — APPROVED** [review]

**Comment by rafayet-monon** [comment]
> > Have you considered using `eslint`?
> 
> I think it has already included by default as it has some configs in package.json 🤔 I can also run the command. But I think I should add a simple github action to run the format check and eslint again 💭 Can I do it in another PR?

@rosle we can do it in another PR. But I can suggest some eslint plugin to make life easier -
- eslint-config-prettier
- eslint-plugin-import
- eslint-plugin-jsx-a11y
- eslint-plugin-react
- eslint-plugin-react-hooks

Please consider adding these when you work on the PR 😄

**Pull request merged**